generator/api/src/main/java/io/fabric8/maven/generator/api/support/BaseGenerator.java

```diff
@@ -121,8 +122,8 @@ protected String getAlias() {
     }
 
     protected boolean shouldAddDefaultImage(List<ImageConfiguration> configs) {
-        String merge =  getConfigWithSystemFallbackAndDefault(Config.alias, "fabric8.generator.alias", "false");
-        return !containsBuildConfiguration(configs) || Boolean.parseBoolean(merge);
+        String merge =  getConfigWithSystemFallbackAndDefault(Config.alias, "fabric8.generator.alias", "");
+        return !containsBuildConfiguration(configs) || Strings.isNotBlank(merge);
     }
```

Comment:
I just added https://github.com/fabric8io/fabric8-maven-plugin/pull/420 to use a more explicit boolean condition; using the generator property `enabled` on java generator / spring boot generator to force the generator to be enabled even if there's already a docker image configured